package/endpoint_adapter/src/endpoint_adapter_can.c

```diff
@@ -35,23 +35,28 @@
 #define BUFFER_SIZE (4096)
 static uint8_t pipe_read_buffer[BUFFER_SIZE];
 
+static bool is_j1939 = false;
+
 static u32 can_filter;
 
 typedef struct {
   int can_read_fd;
   int read_pipe_fd;
+  const char *framer;
 } read_thread_context_t;
 
 typedef struct {
   int can_write_fd;
   int write_pipe_fd;
+  const char *framer;
 } write_thread_context_t;
 
 static void *can_read_thread_handler(void *arg)
 {
   piksi_log(LOG_DEBUG, "CAN read thread starting...");
   read_thread_context_t *pctx = (read_thread_context_t *)arg;
 
+  u8 full_frame[12];
```

Comment:
Declare a constant, or use some `sizeof` operators?  Or just copy in the `#define`s from the j1939 protocol package:
```c
#define J1939_HEADER_LENGTH 4
#define J1939_DATA_LENGTH 8
```